openfisca_aotearoa/variables/live_organ_donor_compensation.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+# Import from openfisca-core the common python objects used to code the legislation in OpenFisca
+from openfisca_core.model_api import *
+# Import the entities specifically defined for this tax and benefit system
+from openfisca_aotearoa.entities import Person
+from numpy import round
+
+class weekly_compensation_before_tax(Variable):
+    """
+    Calculate the weekly compensation based on employee earnings over the last 52 weeks
+    and self-employed earnings in most recent tax year.
+    This relies on the user totalling the earnings over the last 52 weeks and the earnings period in weeks.
+    Ideally a user would be able to enter the earnings per week (over a period) and the totals would then
+    be calculated.
+    """
+    value_type = float
+    entity = Person
+    definition_period = YEAR # TODO - determine whether we need to get WEEK to work
+    label = u"The amount payable as compensation per week before tax"
+    # note that this also implements http://www.legislation.govt.nz/act/public/2016/0096/latest/whole.html?search=ts_act%40bill%40regulation%40deemedreg_live+organ_resel_25_a#DLM6965116
+    reference = "http://www.legislation.govt.nz/act/public/2016/0096/latest/whole.html?search=ts_act%40bill%40regulation%40deemedreg_live+organ_resel_25_a#DLM6965123"
+
+    def formula(persons, period, parameters):
+        earnings_amount = persons('sum_of_earnings_in_last_52_weeks', period) 
+        compensation_amount = persons('sum_of_earnings_during_compensation_period_in_last_52_weeks', period)
+        earnings_excluding_compensation_amount = earnings_amount - compensation_amount
+
+        earnings_period = persons('earnings_period_in_weeks', period)
+        compensation_period = persons('compensation_period_in_weeks', period)
+        earnings_excluding_compensation_period = earnings_period - compensation_period
+
+        self_employed_earnings_in_most_recent_tax_year = persons('sum_of_self_employed_earnings_in_most_recently_completed_tax_year', period)
+        most_recent_tax_year_period = persons('number_of_weeks_in_most_recently_completed_tax_year', period)
+
+        employee_weekly_earnings = earnings_excluding_compensation_amount / earnings_excluding_compensation_period
+        self_employed_weekly_earnings = self_employed_earnings_in_most_recent_tax_year / most_recent_tax_year_period
+        
+        return round(employee_weekly_earnings + self_employed_weekly_earnings, 2)
+
+class kiwisaver_employee_deduction(Variable):
+    value_type = float
```

Comment:
This method might be better in a KiwiSaver specific module, so that it picks up any changes to KiwiSaver.

It's also largely copy and paste from above. It should utilise the method from above to calculate the basic compensation and then calculate KiwiSaver on that amount.

It is also missing concepts such as KiwiSaver holiday period.